src/Material/Options.elm

```diff
@@ -135,24 +175,44 @@ collect1' options acc =
     Class x -> { acc | classes = x :: acc.classes }
     CSS x -> { acc | css = x :: acc.css }
     Attribute x -> { acc | attrs = x :: acc.attrs }
+    Internal x -> { acc | internal = x :: acc.internal }
     Many options -> List.foldl collect1' acc options
     Set _ -> acc 
+    Listener event options decoder ->
+      { acc | listeners = (event, (decoder, options)) :: acc.listeners}
+    Lift m ->
+      { acc | dispatch = Just m}
     None -> acc
 
 
 collect' : List (Property c m) -> Summary () m 
 collect' = 
-  List.foldl collect1' (Summary [] [] [] ())
+  List.foldl collect1' (Summary [] [] [] [] [] Nothing ())
 
 
 addAttributes : Summary c m -> List (Attribute m) -> List (Attribute m)
-addAttributes summary attrs = 
-  List.append
-    attrs
-    (  Html.Attributes.style summary.css 
-    :: Html.Attributes.class (String.join " " summary.classes) 
-    :: summary.attrs
-    )
+addAttributes summary attrs =
+  let
```

Comment:
Again, maybe check what comes out in elm.js when you use let—I seem to recall elm generates a closure because it doesn't want to rely on whatever version of javascript it is that gives us a let construct.